com.unity.shadergraph/Editor/Data/Nodes/Channel/SwizzleNode.cs

```diff
@@ -27,146 +26,174 @@ public override bool hasPreview
             get { return true; }
         }
 
-        public sealed override void UpdateNodeAfterDeserialization()
-        {
-            AddSlot(new DynamicVectorMaterialSlot(InputSlotId, kInputSlotName, kInputSlotName, SlotType.Input, Vector4.zero));
-            AddSlot(new Vector4MaterialSlot(OutputSlotId, kOutputSlotName, kOutputSlotName, SlotType.Output, Vector4.zero));
-            RemoveSlotsNameNotMatching(new[] { InputSlotId, OutputSlotId });
-        }
-
-        static Dictionary<TextureChannel, string> s_ComponentList = new Dictionary<TextureChannel, string>
-        {
-            {TextureChannel.Red, "r" },
-            {TextureChannel.Green, "g" },
-            {TextureChannel.Blue, "b" },
-            {TextureChannel.Alpha, "a" },
-        };
-
         [SerializeField]
-        TextureChannel m_RedChannel;
+        string _maskInput = "xyzw";
 
-        [ChannelEnumControl("Red Out")]
-        public TextureChannel redChannel
+        [TextControl("Mask:")]
+        public string maskInput
         {
-            get { return m_RedChannel; }
+            get { return _maskInput; }
             set
             {
-                if (m_RedChannel == value)
+                if (_maskInput.Equals(value))
                     return;
-
-                m_RedChannel = value;
-                Dirty(ModificationScope.Node);
+                _maskInput = value;
+                UpdateNodeAfterDeserialization();
+                Dirty(ModificationScope.Topological);
             }
         }
 
-        [SerializeField]
-        TextureChannel m_GreenChannel;
+        public string convertedMask;
 
-        [ChannelEnumControl("Green Out")]
-        public TextureChannel greenChannel
+        public bool ValidateMaskInput(int InputValueSize)
         {
-            get { return m_GreenChannel; }
-            set
-            {
-                if (m_GreenChannel == value)
-                    return;
+            _maskInput = _maskInput.ToLower();
```

Comment:
I like the ToLower() call since it lets users not care - artists hate case sensitivity. However, this overwrites their chosen case when they save and reload. Can we keep the lower case version inside our own code (just for validation and actually writing the shader)?